### ISSUE TITLE
Parsing error with async generators

### ISSUE THREAD

**Issue by ice-blaze** [eslint/eslint#10096]
Eslint doesn't handle async generators. Here an example where we got a `parsing error` on the `*`.

```javascript
const foo = async function *(bar) {
    yield bar
}
```
Even with the right parser.
```javasript
"parserOptions": {
    "ecmaVersion": 2017,
},
```

**Comment by not-an-aardvark** [comment]
Hi,

Async generators are not part of ES2017. To enable them, you should use `parserOptions: { ecmaVersion: 2018 }`.

**Comment by ice-blaze** [comment]
It works :) Sorry I feel bad I'm sure I tried it...
